src/envoy/utils/utils.cc

```diff
@@ -90,20 +90,27 @@ bool GetDestinationUID(const envoy::api::v2::core::Metadata& metadata,
   return true;
 }
 
-bool GetSourceUser(const Network::Connection* connection, std::string* user) {
+bool GetPrincipal(const Network::Connection* connection, bool peer,
+                  std::string* principal) {
   if (connection) {
     Ssl::Connection* ssl = const_cast<Ssl::Connection*>(connection->ssl());
     if (ssl != nullptr) {
-      std::string result = ssl->uriSanPeerCertificate();
-      if (result.empty()) {  // empty source user is not allowed
+      std::string result;
+      if (peer) {
+        result = ssl->uriSanPeerCertificate();
```

Comment:
whats the content of this? SAN or the cert itself? Very badly named function

Reply:
This is straight from OpenSSL library. You are welcome to tell them that their naming is bad...